src/SqlAccountRestAPI/Core/SqlAccountFactory.cs

```
@@ -14,33 +15,20 @@ public dynamic GetInstance(bool autoLogin = true)
         {
             try
             {
-                var isLoginTask = Task.Run(() => _app.IsLogin());
-
-                if (!isLoginTask.Wait(TimeSpan.FromSeconds(10)))
+                var comChecker = SystemHelper.IsComObjectResponsive(() => _app.IsLogin, TimeSpan.FromSeconds(5));
+                if (!comChecker)
                 {
-                    throw new TimeoutException("The login process took too long.");
+                    throw new Exception("COM object is not responsive.");
                 }
-
-                // The app must be logined
-                if (!isLoginTask.Result)
-                {
-                    throw new InvalidOperationException("Login failed: _app.IsLogin() returned false.");
+                if(!_app.IsLogin){
+                    throw new Exception("App is not logged in.");
                 }
-
                 return _app;
             }
-            catch (TimeoutException)
-            {
-                Console.WriteLine("Timeout: Login process exceeded 10 seconds.");
-                _app = null;
-            }
-            catch (InvalidOperationException ex)
-            {
-                Console.WriteLine(ex.Message);
-                _app = null;
-            }
-            catch (COMException)
+            catch (Exception e)
             {
+                Console.WriteLine(e.Message);
```

không có try catch rồi console kiểu này. Cùng lắm thì 

Catch (Exception e) { _app = null; throw; }. 

Fail Fast Principle anh đã nói rất nhiều lần rồi mà?